tests/scripted/check-namespace

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/python3
+
+# This script checks the code for namespace related issues
+# Note: This script is not perfect. It can not parse all LMMS files,
+#       and does not contain a complete C++ parser. If you encounter
+#       difficulties with this tests, it could be a fault of your
+#       changes, but it could also be a fault of this script.
+
+import re
+import subprocess
+from pathlib import Path
+
+# global variables
+
+errors = 0
+
+
+# functions
+
+def caption(my_str):
+    print(f'\n# {my_str}\n')
+
+
+def error(where, my_str):
+    global errors
+    errors += 1
+    print(f'Error: {where}: {my_str}')
+
+
+# if a string makes a classname, we need to check if that class is in the source
+# however, some of these strings name classes that are not from LMMS, so these can be ignored for such checks:
+def is_our_class(classname: str) -> int:
+    return classname[0] != 'Q'  # Qt classes
+
+
+# prepare some variables
+
+if not Path('.gitmodules').is_file():
+    print('You need to call this script from the LMMS top directory')
+    exit(1)
+
+result = subprocess.run(['git', 'ls-files', '*.[ch]', '*.[ch]pp', ':!tests/*'],
+                        capture_output=True, text=True, check=True)
+exclude_files = {
+    # too complex
+    'include/AudioPortAudio.h',
+    'src/core/audio/AudioPortAudio.cpp',
+    'include/RemotePluginBase.h',
+    'plugins/VstBase/RemoteVstPlugin.cpp',
+    'plugins/ReverbSC/ReverbSC.h',
+    # not ours:
+    'include/ladspa.h'
+    # f.startswith('plugins/LadspaEffect')
+    # f.__contains__('portsmf')
+}
+known_no_namespace_lmms = {
+    # main.cpp
+    'src/core/main.cpp',
+    # nothing to set under a namespace
+    'include/debug.h',
+    'include/versioninfo.h',
+    'plugins/CarlaBase/CarlaConfig/config.h',
+    'plugins/CarlaBase/DummyCarla.cpp',
+    # unclear why it has no namespace
+    'plugins/ZynAddSubFx/RemoteZynAddSubFx.cpp',
+    # not our code
+    'include/aeffectx.h',
+    'include/fenv.h',
+    'plugins/Sf2Player/fluidsynthshims.h',
+    'plugins/MidiExport/MidiFile.hpp',
+    'plugins/ReverbSC/base.c',
+    'plugins/ReverbSC/base.h',
+    'plugins/ReverbSC/dcblock.c',
+    'plugins/ReverbSC/dcblock.h',
+    'plugins/ReverbSC/revsc.c',
+    'plugins/ReverbSC/revsc.h'
+}
+
+files = [Path(f) for f in result.stdout.splitlines() if
+            f not in exclude_files
+            and not f.startswith('plugins/LadspaEffect/calf')
```

Comment:
I'd put the files filtered here into `exclude_files` for readability.

Reply:
I pushed a new version, can you please check if it's OK?